tiledb/ml/models/sklearn.py

```diff
@@ -39,30 +39,72 @@ def save(self, *, update: bool = False, meta: Optional[Meta] = None) -> None:
             target location.
         :param meta: Extra metadata to save in a TileDB array.
         """
+        if self.artifact is None:
+            raise RuntimeError("Model is not initialized")
+
         # Serialize model
         serialized_model = self._serialize_model()
 
         # Create TileDB model array
         if not update:
-            self.__create_array()
+            self._create_array(fields=["model"])
+
+        self._write_array(model_params={"model": serialized_model})
 
-        self._write_array(serialized_model=serialized_model, meta=meta)
+        if meta:
+            self._write_model_metadata(meta=meta)
 
     def load(self, *, timestamp: Optional[Timestamp] = None) -> BaseEstimator:
         """
-        Load a Sklearn model from a TileDB array.
+        Load switch, i.e, decide between __load (TileDB-ML<=0.8.0) or __load_v2 (TileDB-ML>0.8.0).
 
+        Load a Sklearn model from a TileDB array.
         :param timestamp: Range of timestamps to load fragments of the array which live
             in the specified time range.
         :return: A Sklearn model object.
         """
         # TODO: Change timestamp when issue in core is resolved
 
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
+            # Check if we try to load models with the old 1-cell schema.
+            if model_array.schema.domain.size < np.iinfo(np.uint64).max - 1025:
+                return self.__load(
+                    timestamp=timestamp,
+                )
+            return self.__load_v2(
+                timestamp=timestamp,
+            )
+
+    def __load(self, *, timestamp: Optional[Timestamp] = None) -> BaseEstimator:
+        """
+        Load a Sklearn model from a TileDB array.
+        """
         model_array = tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp)
         model_array_results = model_array[:]
         model = pickle.loads(model_array_results["model_params"].item(0))
         return model
 
+    def __load_v2(self, *, timestamp: Optional[Timestamp] = None) -> BaseEstimator:
+        """
+        Load a Sklearn model from a TileDB array.
+        """
+
+        with tiledb.open(self.uri, ctx=self.ctx, timestamp=timestamp) as model_array:
+            model_meta = dict(model_array.meta.items())
```

Comment:
No need to copy model_array.meta to a new dict, it supports the necessary dict methods